src/utils/fetchBlocks.ts

```diff
@@ -0,0 +1,82 @@
+import { ethers } from 'ethers'
+import Block from '../models/block.model'
+import fetch from 'node-fetch'
+import { logger } from './logger'
+
+const providerURL = process.env.PROVIDER_URL || 'https://rpc.gaiaxtestnet.oceanprotocol.com' // or use your local node 'http://localhost:8545'
+const provider = new ethers.providers.JsonRpcProvider(providerURL)
+
+function getLatestBlockNumber() {
+  return fetch('https://exchangelog.minimal-gaia-x.eu/chain-blocks', {
+    credentials: 'omit',
+    headers: {
+      'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:98.0) Gecko/20100101 Firefox/98.0',
+      Accept: 'application/json, text/javascript, */*; q=0.01',
+      'Accept-Language': 'en-US,en;q=0.5',
+      'X-Requested-With': 'XMLHttpRequest',
+      'Sec-Fetch-Dest': 'empty',
+      'Sec-Fetch-Mode': 'cors',
+      'Sec-Fetch-Site': 'same-origin',
+      'Sec-GPC': '1',
+      'Cache-Control': 'max-age=0'
+    },
+    referrer: 'https://exchangelog.minimal-gaia-x.eu/',
+    method: 'GET',
+    mode: 'cors'
+  })
+    .then(response => response.json())
+    .then(data => data.blocks[0].block_number)
+}
```

Comment:
issue(blocking): There is an easier way to do this without relying on the exchangelog service. You can use etherjs here with `await provider.getBlock('latest')`. This contains also the block number.